hpx/state.hpp

```diff
@@ -18,11 +18,21 @@ namespace hpx
 {
     enum state
     {
-        initialized = BOOST_BINARY_U(000),
-        starting    = BOOST_BINARY_U(001),
-        running     = BOOST_BINARY_U(010),
-        stopping    = BOOST_BINARY_U(100),
-        terminating = BOOST_BINARY_U(101)
```

Comment:
With these gone, `<boost/utility/binary.hpp>` no longer needs to be included here.